extension/src/extension.ts

```diff
@@ -125,6 +118,22 @@ export async function activate(ctx: sourcegraph.ExtensionContext): Promise<void>
                     endRegex: /\*\//,
                 },
             },
+            filterDefinitions: ({ filePath, fileContent, results }) => {
+                const imports = fileContent
+                    .split('\n')
+                    .map(line => {
+                        // Matches the import at index 1
+                        const match = /from ['"](.*)['"];?$/.exec(line) || /require\(['"](.*)['"]\);?$/.exec(line)
+                        return match ? match[1] : undefined
+                    })
+                    .filter((x): x is string => Boolean(x))
+
+                const filteredResults = results.filter(result =>
+                    imports.some(i => path.join(path.dirname(filePath), i) === result.file.replace(/\.[^/.]+$/, ''))
```

Comment:
`result.file.replace(/\.[^/.]+$/, '')` is this `path.basename()`?

`path.join(path.dirname(filePath), i)` should this be `path.resolve()` to resolve `..`?

Reply:
`result.file.replace(/\.[^/.]+$/, '')` gets rid of the extension (from https://stackoverflow.com/a/4250408/2061958)

`path.join` already resolves `..`:

```
> path.join(path.dirname('foo/bar/baz'), '..')
'foo'
```